addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -153,107 +162,42 @@ public void initialize() {
             discoveryServiceRegistry.addDiscoveryListener(this);
         }
 
-        nonUpnpRemoteControllerJob = scheduler.scheduleWithFixedDelay(this::checkCreateManualConnection, 1, 1,
-                TimeUnit.MINUTES);
+        checkAndCreateServices();
+
+        if (configuration.macAddress == null || configuration.macAddress.isEmpty()) {
+            try {
+                Process proc = Runtime.getRuntime().exec("arping -r -c 1 -C 1 " + configuration.hostName);
+                proc.waitFor();
+                BufferedReader stdInput = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+                configuration.macAddress = stdInput.readLine();
+                getConfig().put(SamsungTvConfiguration.MAC_ADDRESS, configuration.macAddress);
+                logger.info("MAC address of host {} is {}", configuration.hostName, configuration.macAddress);
+
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
Can always be more specific, but as the retrieval of the MAC address via arping is optional as already mentioned and sensitive to external requirement, I think it is better to have a broad safety net here.